tripy/tripy/frontend/utils.py

```diff
@@ -405,6 +405,76 @@ def convert_nontensor_arg(arg, list_index=None):
 
     return impl
 
+def wraps_to_flat_ir_to_func(cls):
+    """
+    Decorator that wraps the to_flat_ir method of a class.
+
+    This decorator creates a FlatIRFunction corresponding to the captured FlatIR operations
+    in the to_flat_ir method. Inputs/outputs to the FlatIRFunction are cloned version of original tensors
+    and maintains a caller and callee relationship to help integrate Function IR subgraph with overall FlatIR.
+
+    Args:
+        cls: The class being decorated.
+
+    Returns:
+        The decorated class with a modified to_flat_ir method.
+    """
+    original_to_flat_ir = cls.to_flat_ir
+
+    @functools.wraps(original_to_flat_ir)
+    def wrapped_to_flat_ir(
+        self, inputs: List["FlatIRTensor"], outputs: List["FlatIRTensor"], flat_ir: "FlatIR"
```

Comment:
Seems like the `flat_ir` parameter is not used? How does the function get added to the flat_ir?

Reply:
And if it's not used, we can remove the entire `is_to_flat_ir_wrapped` logic and use `to_flat_ir` directly instead of wrapping it with `convert_to_flat_ir`

Reply:
I have addressed this by adding `make_function` decorator.